### PR TITLE
Added dropdown in booking page assets management

### LINKED ISSUE #1368: [Feature request]: Accordion support for visualising kits on booking view
### Contact Details

_No response_

### Is your feature request related to a problem? Please describe?

We heavily use kits for bookings. We do this for overviewing our booked assets. At the moment, when we add several kits our visualisation gets long and we would prefer to see the kit added and be able to compact it / collapse it on demand. 

This is not applicable for the PDF generation. There the visual is good. 

### Describe the solution you'd like

Screenshot 2024-10-24 at 12 20 56

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vikashsprem**
Closes #1368
/claim #1368

https://github.com/user-attachments/assets/4f4e2754-5909-4f30-a217-c15eaf55d9f5

### CODE REVIEW

**Comment by vikashsprem** [comment]
@carlosvirreira Please review.

File: app/components/booking/booking-assets-column.tsx [hunk 1]
```diff
@@ -60,6 +61,16 @@ export function BookingAssetsColumn() {
     [items]
   );
 
+  const [expandedKits, setExpandedKits] = useState<Record<string, boolean>>({});
+  const [categoryVisibility, setCategoryVisibility] = useState<number>(0);
+  const toggleKitExpansion = (kitId: string) => {
+    setExpandedKits((prev: any) => ({
```
Comment (DonKoko) [inline app/components/booking/booking-assets-column.tsx]: No need to use any. You have the type defined above.

File: app/components/booking/booking-assets-column.tsx [hunk 2]
```diff
@@ -60,6 +61,16 @@ export function BookingAssetsColumn() {
     [items]
   );
 
+  const [expandedKits, setExpandedKits] = useState<Record<string, boolean>>({});
+  const [categoryVisibility, setCategoryVisibility] = useState<number>(0);
```
Comment (DonKoko) [inline app/components/booking/booking-assets-column.tsx]: Can you please annotate this, so when another developer opens they can know directly what `categoryVisibility` refers to.

File: app/components/booking/booking-assets-column.tsx [hunk 3]
```diff
@@ -181,20 +193,34 @@ export function BookingAssetsColumn() {
...
-                          {assets.map((asset) => (
-                            <ListItem key={asset.id} item={asset}>
-                              <ListAssetContent
-                                item={asset as AssetWithBooking}
-                              />
-                            </ListItem>
-                          ))}
+                          {isExpanded &&
+                            assets.map((asset) => (
+                              <ListItem key={asset.id} item={asset}>
```
Comment (DonKoko) [inline app/components/booking/booking-assets-column.tsx]: Can you please animate them with framer. 1 thing you might have to slightly adjust the `ListItem` to use use `motion.tr` and receive optional animation props. Needs some testing I am not 100% sure exactly what the right approach would be.

File: app/components/booking/booking-assets-column.tsx [hunk 4]
```diff
@@ -181,20 +193,34 @@ export function BookingAssetsColumn() {
...
-                            <Td className="pr-4 text-right">
-                              {(!isBase && isDraft) || isReserved ? (
-                                <KitRowActionsDropdown kit={kit} />
-                              ) : null}
+                            <Td className="pr-4 text-right align-middle">
+                              <div className="flex items-center justify-end gap-5">
+                                <Button
+                                  onClick={() => toggleKitExpansion(kit.id)}
+                                  variant="link"
+                                  className="text-center font-bold"
```
Comment (DonKoko) [inline app/components/booking/booking-assets-column.tsx]: Please use `text-gray-600` so the icon inherits it and its not orange.

File: app/components/booking/booking-assets-column.tsx [hunk 5]
```diff
@@ -181,20 +193,34 @@ export function BookingAssetsColumn() {
...
+                                >
+                                  {isExpanded ? (
+                                    <ChevronUpIcon className="size-6" />
```
Comment (DonKoko) [inline app/components/booking/booking-assets-column.tsx]: This works but its not how we usually do it. You can just use `ChevronDownIcon` and then use `tw()` helper to dynamically add a rotate class based on `isExpanded`. That way you can also add a css animation to it.

**Review by DonKoko — CHANGES_REQUESTED** [review]
Great first PR. I have added some feedback with some changes needed to make it more fitting to our normal coding style in shelf.

File: app/components/booking/booking-assets-column.tsx [hunk 6]
```diff
@@ -181,20 +201,42 @@ export function BookingAssetsColumn() {
...
                             </Td>
                           </ListItem>
+                            {isExpanded &&
+                            assets.map((asset) => (
+                                <ListItem
+                                motionProps={{ initial: { opacity: 0, y: -10 },
+                                animate: { opacity: 1, y: 0 },
+                                exit: { opacity: 0, y: 10 },
+                                transition: { duration: 0.3, ease: "easeInOut" },
+                                whileHover: { scale: 1.02 },
```
Comment (DonKoko) [inline app/components/booking/booking-assets-column.tsx]: This causes a bug in the UI. Please remove the tap and hover effects.

**Review by DonKoko — CHANGES_REQUESTED** [review]
- [ ] When items are folded its strange they dont have a border between them:
Screenshot 2024-10-25 at 14 42 28
- [ ] Same when I hover an item. We need to address those cases to make the UI consistent
Screenshot 2024-10-25 at 14 42 36
- [ ] Hover of arrow should be `text-gray-500`. Now it becomes orange which is wrong.
Screenshot 2024-10-25 at 14 43 16
- [ ] Lets make the default state to be "open"

**Review by DonKoko — APPROVED** [review]

**Comment by DonKoko** [comment]
@vikashsprem I just had to make some changes to the borders. Thanks for the PR.

**Comment by carlosvirreira** [comment]
Nice, I've seen it on live. Good improvement. Thank you both!

**Pull request merged**